File: src/test/java/io/icednut/algorithm/exercise/trie/leet440/SolutionTest.java
package io.icednut.algorithm.exercise.trie.leet440;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

class SolutionTest {

    @Test
    void test01() {
        var solution = new Solution();

        int output = solution.findKthNumber(13, 2);

        assertEquals(10, output);
    }

    @Test
    void test02() {
        var solution = new Solution();

        int output = solution.findKthNumber(1, 1);

        assertEquals(1, output);
    }

    @Test
    void test03() {
        var solution = new Solution();

        int output = solution.findKthNumber(2, 2);

        assertEquals(2, output);
    }

    @Test
    void test04() {
        var solution = new Solution();

        int output = solution.findKthNumber(7747794, 5857460);

        assertEquals(6271710, output);
    }
}

File: src/test/kotlin/io/icednut/algorithm/exercise/linkedlist/leet3217/SolutionTest.kt
package io.icednut.algorithm.exercise.linkedlist.leet3217

import org.junit.jupiter.api.Test
import kotlin.test.assertEquals

class SolutionTest {

    @Test
    fun test01() {
        val solution = Solution()

        val root = ListNode(1)
        val b = ListNode(2)
        val c = ListNode(3)
        val d = ListNode(4)
        val e = ListNode(5)
        root.next = b
        b.next = c
        c.next = d
        d.next = e

        val output = solution.modifiedList(
            nums = intArrayOf(1, 2, 3),
            head = root
        )

        assertEquals(d, output)
    }

    @Test
    fun test02() {
        val solution = Solution()

        val root = ListNode(1)
        val b = ListNode(2)
        val c = ListNode(1)
        val d = ListNode(2)
        val e = ListNode(1)
        val f = ListNode(2)
        root.next = b
        b.next = c
        c.next = d
        d.next = e
        e.next = f

        val output = solution.modifiedList(
            nums = intArrayOf(1),
            head = root
        )

        assertEquals(b, output)
        assertEquals(d, output?.next)
        assertEquals(f, output?.next?.next)
    }

    @Test
    fun test03() {
        val solution = Solution()

        val root = ListNode(4)
        val b = ListNode(8)
        val c = ListNode(9)
        val d = ListNode(5)
        val e = ListNode(2)
        root.next = b
        b.next = c
        c.next = d
        d.next = e

        val output = solution.modifiedList(
            nums = intArrayOf(3, 5, 9, 8, 7),
            head = root
        )

        assertEquals(root, output)
        assertEquals(e, output?.next)
    }

    @Test
    fun test04() {
        val solution = Solution()

        val root = ListNode(1)
        val b = ListNode(2)
        val c = ListNode(3)
        val d = ListNode(4)
        val e = ListNode(5)
        root.next = b
        b.next = c
        c.next = d
        d.next = e

        val output = solution.modifiedList(
            nums = intArrayOf(2, 3, 4, 5),
            head = root
        )

        assertEquals(root, output)
        assertEquals(null, output?.next)
    }
}